src/Adapter/Cart/CommandHandler/UpdateCartDeliverySettingsHandler.php

```diff
@@ -139,4 +147,97 @@ private function createCartRule(Cart $cart, string $backOfficeOrderCode): CartRu
 
         return $freeShippingCartRule;
     }
+
+    /**
+     * This method works as follows:
+     * 1. if free shipping should be enabled, enable it
+     * 2. if free shipping should not be enabled and cart already does not have free shipping, do nothing
+     * 3.if free shipping should not be enabled and cart has free shipping, disable it
+     *
+     * @param Cart $cart
+     * @param UpdateCartDeliverySettingsCommand $command
+     *
+     * @throws CannotDeleteCartRuleException
+     */
+    protected function handleFreeShippingOption(Cart $cart, UpdateCartDeliverySettingsCommand $command): void
+    {
+        $backOfficeOrderCode = sprintf('%s%s', CartRule::BO_ORDER_CODE_PREFIX, $cart->id);
+
+        $freeShippingCartRule = $this->getCartRuleForBackOfficeFreeShipping($backOfficeOrderCode);
+
+        $freeShippingShouldBeEnabled = $command->allowFreeShipping();
+
+        // Step 1
+        if ($freeShippingShouldBeEnabled) {
+            if (null === $freeShippingCartRule) {
+                // there is not yet a 'free shipping' cart rule available in the system so we create it
+                $freeShippingCartRule = $this->createCartRule($cart, $backOfficeOrderCode);
+            }
+            $cart->addCartRule((int) $freeShippingCartRule->id);
+
+            return;
+        }
+
+        if (null === $freeShippingCartRule) {
+            return;
+        }
+
+        $cart->removeCartRule((int) $freeShippingCartRule->id);
+
+        return;
```

Comment:
```suggestion
        try {
            if (false === $freeShippingCartRule->delete()) {
                throw new CannotDeleteCartRuleException(sprintf('Failed deleting cart rule #%s', $freeShippingCartRule->id));
            }
        } catch (PrestaShopException $e) {
            throw new CartRuleException(sprintf('An error occurred when trying to delete cart rule #%s', $freeShippingCartRule->id));
        }
```

you removed a part of the function